docs/src/modules/ROOT/pages/integration/integration.adoc

```diff
@@ -271,22 +271,743 @@ To use Timefold Solver with Quarkus, read the xref:quickstart/quarkus/quarkus-qu
+[source,kotlin]
+----
+@Path("path")
+class Resource {
+
+    @Inject
+    var solverConfig:SolverConfig?
+
+    @Inject
+    var solverFactory:SolverFactory<Timetable>?
+
+    @Inject
+    var solverManager:SolverManager<Timetable, String>?
+
+    @Inject
+    var simpleSolutionManager:SolutionManager<Timetable, SimpleScore>? // <1>
+
+    @Inject
+    var simpleScoreManager:ScoreManager<Timetable, SimpleScore>? // <1>
+
+    @Inject
+    var constraintVerifier:ConstraintVerifier<TimetableConstraintProvider, Timetable>? = null
+
+    ...
+}
+----
+<1> The following score types are available: `SimpleScore`, `SimpleLongScore`, `SimpleBigDecimalScore`, `HardSoftScore`, `HardSoftLongScore`, `HardSoftBigDecimalScore`, `HardMediumSoftScore`, `HardMediumSoftLongScore`, `HardMediumSoftBigDecimalScore`, `BendableScore`, `BendableLongScore`, and `BendableBigDecimalScore`.
+--
+====
+
+Timefold provides all the necessary resources for problem-solving and analysis. However, it is still possible to manually create and override the default managed resources. To create a custom `SolverManager`, use the `SolverFactory` resource.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+package org.acme.employeescheduling.rest;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Default;
+import jakarta.ws.rs.Produces;
+
+import ai.timefold.solver.core.api.solver.SolverFactory;
+import ai.timefold.solver.core.api.solver.SolverManager;
+import ai.timefold.solver.core.config.solver.SolverManagerConfig;
+
+import org.acme.employeescheduling.domain.EmployeeSchedule;
+
+@ApplicationScoped
+public class BeanProducer {
+
+    @Produces
+    @Default
+    public SolverManager<Timetable, String> overrideSolverManager(SolverFactory<Timetable> solverFactory) {
+        SolverManagerConfig solverManagerConfig = new SolverManagerConfig();
+        return SolverManager.create(solverFactory, solverManagerConfig);
+    }
+}
+----
+--
+Kotlin::
++
+--
+[source,kotlin]
+----
+package org.acme.employeescheduling.rest
+
+import ai.timefold.solver.core.api.solver.SolverFactory
+import ai.timefold.solver.core.api.solver.SolverManager
+import ai.timefold.solver.core.config.solver.SolverManagerConfig
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.enterprise.inject.Default
+import jakarta.ws.rs.Produces
+import org.acme.kotlin.schooltimetabling.domain.Timetable
+
+@ApplicationScoped
+class BeanProducer {
+
+    @Produces
+    @Default
+    fun overrideSolverManager(solverFactory: SolverFactory<Timetable>?): SolverManager<Timetable, String> {
+        val solverManagerConfig = SolverManagerConfig()
+        return SolverManager.create(solverFactory, solverManagerConfig)
+    }
+}
+----
+--
+====
+
+[NOTE]
+====
+Consider using xref:#integrationWithQuarkusMultiStage[Multi-stage planning] for multiple solver configurations instead of manually creating resources.
+====
+
+[#integrationWithQuarkusMultiStage]
+=== Multi-stage planning
```

Comment:
I am not entirely sure why we're selling this as multi-stage planning. This feature can be used for other uses as well. It is a generic feature.

Personally, I'd remove references to multi-stage, turn it into a generic chapter on injecting multiple instances of `SolverManager`, and maybe mention in a note that this feature could be used for multi-stage planning.